buildstockbatch/gcp/gcp.py

```diff
@@ -489,6 +487,49 @@ def show_jobs(self):
             logger.info(f"No existing Cloud Run jobs match {self.postprocessing_job_name}")
         logger.info(f"See all Cloud Run jobs at https://console.cloud.google.com/run/jobs?project={self.gcp_project}")
 
+    def get_existing_batch_job(self):
+        client = batch_v1.BatchServiceClient()
+        try:
+            job = client.get_job(batch_v1.GetJobRequest(name=self.gcp_batch_job_name))
+            return job
+        except exceptions.NotFound:
+            return None
+
+    def get_existing_postprocessing_job(self):
+        jobs_client = run_v2.JobsClient()
+        try:
+            job = jobs_client.get_job(name=self.postprocessing_job_name)
+            return job
+        except exceptions.NotFound:
+            return False
+
+    def check_for_existing_jobs(self, pp_only=False):
+        """If there are existing jobs with the same ID as this job, logs them as errors and returns True.
+
+        Otherwise, returns False.
+
+        :param pp_only: If true, only check for the post-processing job.
+        """
+        if pp_only:
+            existing_batch_job = None
+        else:
+            if existing_batch_job := self.get_existing_batch_job():
+                logger.error(
+                    f"A Batch job with this ID ({self.job_identifier}) already exists "
+                    f"(status: {existing_batch_job.status.state.name}). Choose a new job_identifier or run with "
+                    "--clean to delete the existing job."
+                )
+        if existing_pp_job := self.get_existing_postprocessing_job():
+            status = "Running"
+            if existing_pp_job.latest_created_execution.completion_time:
+                status = "Completed"
```

Comment:
assign `status` based on the condition rather than assigning and possibly reassigning (i.e., `status = "Completed" if existing_pp_job.latest_created_execution.completion_time else "Running"`).